frontend/packages/kubevirt-plugin/src/components/vms/desktop-viewer-selector.tsx

```diff
@@ -0,0 +1,179 @@
+          -rdp --port [UNIQUE_PORT] --target-port {DEFAULT_RDP_PORT} --type NodePort
+        </li>
+      </ul>
+      Make sure, the VM object has <b>spec.template.metadata.labels</b> set to{' '}
+      <b>
+        {TEMPLATE_VM_NAME_LABEL}: {getName(vm)}
+      </b>
+    </span>
+  </div>
+);
+
+export const DesktopViewerSelector: React.FC<DesktopViewerSelectorProps> = (props) => {
+  const { vm, vmi, guestAgent, rdpServiceManual, vncServiceManual } = props;
+
+  const networks = React.useMemo(() => getVmRdpNetworks(vm, vmi), [vm, vmi]);
+  const networkItems = networks.reduce((result, network) => {
+    result[network.name] = network.name;
+    return result;
+  }, {});
+  const [selectedNetwork, setSelectedNetwork] = React.useState(getDefaultNetwork(networks));
+
+  const onNetworkChange = React.useCallback(
+    (newValue) => {
+      const selected = networks.find((n) => n.name === newValue);
+      setSelectedNetwork(selected);
+    },
+    [networks],
+  );
+
+  let content = null;
+  const networkType = selectedNetwork && selectedNetwork.type;
+  switch (networkType) {
+    case NetworkType.MULTUS:
+      if (!guestAgent) {
+        content = (
+          <Alert variant="warning" isInline title={GUEST_AGENT_WARNING_TITLE}>
+            {GUEST_AGENT_WARNING}
+          </Alert>
+        );
+      } else if (!selectedNetwork || !selectedNetwork.ip) {
+        content = (
+          <Alert variant="warning" isInline title={NO_IP_ADDRESS_TITLE}>{`${NO_IP_ADDRESS} ${
+            selectedNetwork ? selectedNetwork.name : ''
+          }`}</Alert>
+        );
+      } else {
+        const rdp = {
+          address: selectedNetwork.ip,
+          port: DEFAULT_RDP_PORT,
+        };
+        content = <DesktopViewer rdp={rdp} />;
+      }
+      break;
+    case NetworkType.POD:
+      content =
+        rdpServiceManual || vncServiceManual ? (
+          <DesktopViewer rdp={rdpServiceManual} vnc={vncServiceManual} />
+        ) : (
+          <RdpServiceNotConfigured vm={vm} />
+        );
+      break;
+    default:
+      // eslint-disable-next-line no-console
+      console.warn(`Unknown network interface type ${networkType}`);
+  }
+
+  return (
+    <div className="kubevirt-desktop-viewer-selector">
+      <Form isHorizontal>
+        <FormGroup
+          className="kubevirt-desktop-viewer-selector__form-group"
+          fieldId="network-dropdown"
+          label={NIC}
+        >
+          <Dropdown
+            id="network-dropdown"
+            onChange={onNetworkChange}
+            items={networkItems}
+            selectedKey={selectedNetwork ? selectedNetwork.name : undefined}
+            title={SELECT_NETWORK_INTERFACE}
+          />
+        </FormGroup>
+      </Form>
+      {content}
+    </div>
+  );
+};
+DesktopViewerSelector.displayName = 'DesktopViewer';
+
+type DesktopViewerSelectorProps = {
+  vm: VMKind;
+  vmi: VMIKind;
+  guestAgent: boolean;
+  rdpServiceManual: any;
```

Comment:
can you add better types ? for `vncServiceManual` too

Reply:
done